### ISSUE TITLE
timezone issue with raptor test

### ISSUE THREAD

**Issue by tbuckl** [r-transit/tidytransit#128]
from cran: 

See https://cran.r-project.org/web/checks/check_results_tidytransit.html

Note that the failing run was in Europe, where there was a DST 
transition this morning.

Please correct ASAP and before Nov 10 to safely retain the package on CRAN.

**Comment by mpadge** [comment]
That's so impressive that your tests can pick that up!! I'm going to have a look at that code and check out how you managed that. Nice work!

**Comment by polettif** [comment]
I was hoping `hms::hms` doesn't have those timezone quirks. It seems the expected time values in the tests are not converted correctly with [parse_hms](https://github.com/r-transit/tidytransit/blob/bdfd5523b40b8bc4efc44d1396175ffe975f75b0/tests/testthat/test-raptor.R#L136-L143) while the actual calculated values are correct. But then again, the same code snippets above don't throw errors. 

The errors also only happen in tests with `keep = "earliest"` but there's [no time conversion happening there](https://github.com/r-transit/tidytransit/blob/master/R/raptor.R#L203-L206).

So in short, I don't know what's going on. I can't reproduce the errors on my system and hms _should_ store all times in UTC. Maybe it was just a timing issue on the server and will sort itself out...

**Comment by mpadge** [comment]
Ah, i see - it's just your tests. `hms::parse_hms` accepts objects of the `hms` class, and so coerces these as `difftime` objects. These in turn have a [`tz` argument](https://stat.ethz.ch/R-manual/R-devel/library/base/html/difftime.html), which is by default [`Sys.timezone(location = TRUE)`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/timezones.html). All you need to do is hard-code the `tz` in your tests, so that your test lines linked to above of:
``` r
expected = as.numeric(hms::parse_hms(c(
    "stop2"  = "07:05:00", # departure time
    "stop3a" = "07:11:00",
    "stop3b" = "07:11:10",
    "stop4"  = "07:37:00",
    "stop8a" = "07:24:10",
    "stop8b" = "07:24:00"
  )))
```
instead have to be written
``` r
mytz <- "GMT+1" # or whatever
t0 <- strptime ("00:00:00", format = "%H:%M:%S", tz = mytz)
tt <- c ("07:05:00", "07:11:00", "07:11:10", "07:37:00", "07:24:10", "07:24:00")
expected <- difftime (strptime (tt, format = "%H:%M:%S", tz = mytz), t0, units = "secs")
names (expected) <- c ("stop2", "stop3a", "stop3b", "stop4", "stop8a", "stop8b")
```
The problem is `hms`, which is intentionally designed to do "lazy" conversion so that `parse_hms` will only ever assume system time at current locale, and can't be (easily) convinced otherwise. `strptime` grants you the full control you need, so I'd recommend using that instead.

**Comment by polettif** [comment]
Thanks! Glad to hear it's the tests. I decided to simply use the numeric values instead of unnecessarily parsing and converting strings.
